src/agent.c

```diff
@@ -1020,7 +1020,14 @@ int agent_bookkeeping(juice_agent_t *agent, timestamp_t *next_timestamp) {
 			// the corresponding TURN session regularly
 			if (relay_entry) {
 				relay_entry->state = AGENT_STUN_ENTRY_STATE_SUCCEEDED_KEEPALIVE;
-				agent_arm_transmission(agent, relay_entry, TURN_REFRESH_PERIOD);
+#ifdef NO_ATOMICS
+				bool must_arm = !relay_entry->armed;
+#else
+				bool must_arm = !atomic_flag_test_and_set(&relay_entry->armed);
+#endif
+				if (must_arm) {
```

Comment:
This works, but the `armed` flag is only here to allow rearming the keepalives from the agent thread after a send from the user, so the send function does not need to lock the mutex each time. It would be cleaner to refactor a bit the logic here to arm transmission only if the entry state changed.